src/chapters/orm-intro/data-stores.rst

```diff
@@ -0,0 +1,157 @@
+
+Since our data store, ``EventDbContext``, extends ``DbContext``, we have access to all of the methods defined by ``DbContext``. There are quite a few such methods (see `the documentation <https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbcontext?view=efcore-3.1#methods>`_ for details), but we will only use ``SaveChanges``. 
+
+We will make more extensive use of the ``Events`` property of ``EventDbContext``, which is of type ``DbSet``. As mentioned in the previous section, this property allows us to query for object directly from the database. It too `has quite a few methods <https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbset-1?view=efcore-3.1#methods>`_, and we will demonstrate usage of ``Add``, ``Remove``, ``ToList``, and ``Find``.
+
+.. todo: add data store / controller video
+
+Data Stores in the Controller - Text
+-------------------------------------
+
+Adding the Data Store
+^^^^^^^^^^^^^^^^^^^^^
+
+To access data from the database in a controller, we'll need an instance of ``EventDbContext``. Adding the following code to the top of ``EventsController`` will make one available as an instance variable of the controller.
+
+.. sourcecode:: csharp
+   :lineno-start: 17
+
+   private EventDbContext context;
+
+   public EventsController(EventDbContext dbContext)
+   {
+      context = dbContext;
+   }
+
+We can now reference ``context`` anywhere within our controller in order to query the database.
+
+.. index:: dependency injection
+
+.. admonition:: Note
+
+   It may not be obvious how this code works, since we never explicitly call the constructor for ``EventsController``. When we run our application, ASP.NET will call this constructor for us, and will pass in an instance of ``EventDbContext``. This is an example of the concept of **dependency injection**, a topic which is fairly complex and beyond the scope of this course.
+
+Using Data Store Methods
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let's refactor our controller to replace each usage of the in-memory store ``EventData`` with our instance of ``EventDbContext``. 
+
+The first such usage is in the ``Index`` method, which displays a listing of all of the events. We are currently calling ``EventData.getAll()``. To carry out the equivalent action on ``context``, we can use its ``Events`` property. Modify ``Index`` to look like the code below.
+
+.. sourcecode:: csharp
+   :lineno-start: 25
+
+   public IActionResult Index()
+   {
+      List<Event> events = context.Events.ToList();
+
+      return View(events);
+   }
+
+We use the ``ToList`` method of ``context.Events`` (recall that this property is a ``DbSet``) to fetch a list of *all* ``Event`` objects stored in the database.
+
+Our next usage of ``EventData`` that needs to be replaced is in the ``NewEvent`` method. That method is currently calling ``EventData.Add(newEvent)`` to store a newly created ``Event`` object. Replace this line with the following.
```

Comment:
For better or worse, I had changed this on the branches in model validation sections, so the newEvent method is call Add() to process form submission. Sally pointed out that we decided not to do this so maybe to rectify that here we can add a note that we've changed the method name again? or just call it the add method that handles post requests as you do below with delete? :/ srry thats kind of confusing

Reply:
I changed this to "the `Add` method that handles POST requests" but I'm a bit confused. I started with the `enums` branch code which had `NewEvent` as the method name. Are you saying that at some point before that branch the method incorrectly renamed? If so, are we going to go back and fix it? In that case, I can correct it from here going forward.

Reply:
hmm maybe i missed it on enums. i'll make the change there too

Reply:
yeah i hadnt updated it there sorry. i also noticed the constructor chaining i pointed out elsewhere in this pr wasnt on the branch either so i added that too